### PR TITLE
write_env_usage: use proper TOML writer (fix #1681)

### LINKED ISSUE #1681: Pkg.gc not working on windows due to string interpolation
I reported a bug to julia, but I think should be listed here. 

https://github.com/JuliaLang/julia/issues/34781

### PR DESCRIPTION
**Pull request by StefanKarpinski**
The main fix here is to use TOML.print to generate the new content that is appended to the `manifest_usage.toml` log file, which should correctly escape the TOML key. I also moved the early exit test for the `source_file` not existing earlier so that we don't do any work in that case.

### CODE REVIEW

File: src/Types.jl [hunk 1]
```diff
@@ -377,19 +377,22 @@ function Context!(ctx::Context; kwargs...)
...
     touch(usage_file)
 
-    # Don't record ghost usage
-    !isfile(source_file) && return
+    # Write entire new entry to a buffer first
+    buf = IOBuffer()
+    TOML.print(buf, Dict(source_file => [Dict("time" => now())]))
+    entry = String(take!(buf))
 
     # Do not rewrite as do-block syntax (no longer precompilable)
```
Comment (KristofferC) [inline src/Types.jl]: FWIW, we can remove this now since we now can precompile these things.

**Comment by StefanKarpinski** [comment]
This fix probably also applies to 1.0.x, so I'm tagging that here, but I'm not sure how urgent backporting it actually is.

File: src/Types.jl [hunk 2]
```diff
@@ -377,20 +377,22 @@ function Context!(ctx::Context; kwargs...)
...
-    io = open(usage_file, "a")
-    print(io, """
-    [[$(repr(source_file))]]
-    time = $(now())Z
-    """)
-    close(io)
+    # Ensure that log dir exists
+    !ispath(logdir()) && mkpath(logdir())
+
+    # Generate entire as a string first
```
Comment (feanor12) [inline src/Types.jl]: entire -> entry ?

File: src/Types.jl [hunk 3]
```diff
@@ -377,20 +377,22 @@ function Context!(ctx::Context; kwargs...)
...
+    entry = sprint() do io
+        TOML.print(buf, Dict(source_file => [Dict("time" => now())]))
```
Comment (feanor12) [inline src/Types.jl]: buf -> io

**Pull request merged**